rmf_site_editor/src/workspace.rs

```diff
@@ -264,208 +233,268 @@ pub fn dispatch_new_workspace_events(
-                        match Workcell::from_urdf(&urdf) {
-                            Ok(workcell) => {
-                                // Switch state
-                                app_state.set(AppState::WorkcellEditor);
-                                load_workcell.send(LoadWorkcell {
-                                    workcell,
-                                    focus: true,
-                                    default_file,
-                                });
-                                interaction_state.set(InteractionState::Enable);
-                            }
-                            Err(err) => {
-                                error!("Failed converting urdf to workcell {:?}", err);
-                            }
+        }
+        WorkspaceData::WorkcellUrdf(data) => {
+            info!("Importing urdf workcell");
+            let Ok(utf) = std::str::from_utf8(&data) else {
+                error!("Failed converting urdf bytes to string");
+                return;
+            };
+            match urdf_rs::read_from_string(utf) {
+                Ok(urdf) => {
+                    // TODO(luca) make this function return a result and this a match statement
+                    match Workcell::from_urdf(&urdf) {
+                        Ok(workcell) => {
+                            // Switch state
+                            app_state.set(AppState::WorkcellEditor);
+                            load_workcell.send(LoadWorkcell {
+                                workcell,
+                                focus: true,
+                                default_file,
+                            });
+                            interaction_state.set(InteractionState::Enable);
+                        }
+                        Err(err) => {
+                            error!("Failed converting urdf to workcell {:?}", err);
                         }
-                    }
-                    Err(err) => {
-                        error!("Failed loading urdf workcell {:?}", err);
                     }
                 }
-            }
-            WorkspaceData::LoadSite(site) => {
-                app_state.set(AppState::SiteEditor);
-                load_site.send(site);
-                interaction_state.set(InteractionState::Enable);
+                Err(err) => {
+                    error!("Failed loading urdf workcell {:?}", err);
+                }
             }
         }
+        WorkspaceData::LoadSite(site) => {
+            app_state.set(AppState::SiteEditor);
+            load_site.send(site);
+            interaction_state.set(InteractionState::Enable);
+        }
+    }
+}
+
+#[derive(Resource)]
+/// Services that deal with workspace loading
+pub struct WorkspaceLoadingServices {
+    /// Service that spawns an open file dialog and loads a site accordingly.
+    pub load_workspace_from_dialog: Service<(), (), ()>,
```

Comment:
Very minor remark: The third generic argument is for streams which are optional (the absence of streams is represented by an empty tuple). If you just leave it out it'll [fill in the empty tuple automatically](https://github.com/open-rmf/bevy_impulse/blob/3e681a91033427c5ae24995f4e53d19f656f278f/src/service.rs#L68).

So for example this could be `Service<(), ()>`. It's a very minor thing, but I think it's nicer to just see the input and output in the Service signature when there are no streams.

Reply:
Didn't know that! Done https://github.com/open-rmf/rmf_site/pull/233/commits/659da701cabb5f2f56ba45473bf7ec028866d0b1